### PR TITLE
fix #935: Improve check-random-words

### LINKED ISSUE #935: Improve check-random-words
Currently --check-random-words will choose randomly words in the whole json file. Between 2 runs, it happens that we have the same word. I suggest to rename check-random-words to check-words and add a --random option. Also add the ability to check all words in the list.

### PR DESCRIPTION
**Pull request by lasconic**
Fixes #935

### CODE REVIEW

File: wikidict/check_words.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+from . import check_word, render
+
+
+def main(locale: str, count: int, random: bool, offset: str, input: str) -> int:
+    """Entry point."""
+
+    output_dir = Path(os.getenv("CWD", "")) / "data" / locale
+    all_words: List[str] = []
+    if input:
+        with open(input, "r") as fp:
```
Comment (BoboTiG) [inline wikidict/check_words.py:17]:
```suggestion
        with open(input) as fp:
```

File: tests/test_4_check_words.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+
+def test_all(tmp_path):
+    file = tmp_path / "test.json"
+    file.write_text('{"base":""}')
+    with patch.object(render, "get_latest_json_file", return_value=file):
+        assert check_words.main("fr", -1, False, "", "") == 0
+
+
+def test_input(tmp_path):
```
Comment (BoboTiG) [inline tests/test_4_check_words.py:28]: Maybe rename would be a good thing to catch the offset possibilities?
```suggestion
def test_word_offset(tmp_path):
```

File: wikidict/__main__.py [hunk 3]
```diff
@@ -23,8 +23,11 @@
                                 - "data/$LOCALE/dicthtml-$LOCALE.zip": Kobo format.
                                 - "data/$LOCALE/dict-$LOCALE.df": DictFile format.
   --find-templates          DEBUG: Find all templates in use.
-  --check-random-words=N    Get and render N words.
-                            Then compare with the rendering done on the Wiktionary to catch errors.
+  --check-words             Render words, then compare with the rendering done on the Wiktionary to catch errors.
+  --random                  randomly if --random
+  --count=N                 If -1 check all words [default: 100]
+  --offset=M                Offset will remove words before starting.
+  --input=FILENAME          A list of words, one by line
```
Comment (BoboTiG) [inline wikidict/__main__.py:30]: This looks ugly, no?

To be right, those options should be suboptions to `--check-words`, something like:
```suggestion
  --check-words             Render words, then compare with the rendering done on the Wiktionary to catch errors.

                            --random          Check random words.
                            --count=N         If -1 check all words, else check N words [default: 100]
                            --offset=M        Start checking words at offset M if M is a number, else start checking words at word M.
                            --input=FILENAME  A list of words, one by line.
```

I am still not satisfied with those changes though. :thinking:

File: wikidict/check_words.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            return 1
+
+        print(f">>> Loading {file} ...")
+        all_words = list(render.load(file).keys())
+
+    if count == -1:
+        count = len(all_words)
+
+    if offset:
+        if offset.isnumeric():
```
Comment (BoboTiG) [inline wikidict/check_words.py:32]:
```suggestion
        if offset.isnumeric():  # offset = "42"
```

File: wikidict/check_words.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+            offset_int = int(offset)
+            all_words = all_words[offset_int:]
+        else:
```
Comment (BoboTiG) [inline wikidict/check_words.py:35]:
```suggestion
        else:  # offset = "some word"
```

**Review by BoboTiG — COMMENTED** [review]
Well, proposed changes are not blocker. You can take them into account or not :)

And tt is right that parallelizing those checks would be great!

File: wikidict/__main__.py (hunk 3 above)
Comment (lasconic) [reply wikidict/__main__.py:30]: Yes, it's ugly, but I couldn't make it work differently :-/ Your version looks better.

**Review by BoboTiG — APPROVED** [review]

**Pull request merged**